tests/utils.py

```diff
@@ -77,44 +69,75 @@ def clear_data(self):
         CobrancasBBWrapper().reset_data()
 
     @staticmethod
-    def build_response_mock(status_code=200, data=None, content=None):
-        """
-        Cria uma response mockada
-        """
-        response = MagicMock(
-            status_code=status_code, data=data if data else {}, content=content
-        )
+    def __auth_success_201_data(call_count):
+        return {
+            "access_token": f"token_{call_count}",
+            "token_type": "token_type",
+            "expires_in": 600,
+        }
 
-        response.json.return_value = response.data
+    def build_auth_success_response(self, call_count):
+        status = 201
+        headers = self._get_headers()
+        data = json.dumps(self.__auth_success_201_data(call_count))
+        return status, headers, data
 
-        def raise_for_status():
-            # noinspection PyCallByClass
-            requests.Response.raise_for_status(response)
+    @staticmethod
+    def __auth_fail_401_data(call_count):
+        return {
+            "error": f"invalid_client{call_count}",
+            "error_description": "Identificador ou credencial inválidos",
+        }
 
-        response.raise_for_status.side_effect = raise_for_status
-        return response
+    def build_auth_fail_response_401(self, call_count):
+        status = 401
+        headers = self._get_headers()
+        data = json.dumps(self.__auth_fail_401_data(call_count))
+        return status, headers, data
 
-    def set_auth(self):
-        self.mocked_auth_requests.post.reset_mock()
+    def __get_auth_request(self):
+        return self.mock.registered()[0]
 
-        def request_auth(*args, **kwargs):
-            call_count = self.mocked_auth_requests.post.call_count
-            return self.build_response_mock(
-                200,
-                data={
-                    "access_token": f"token_{call_count}",
-                    "token_type": "token_type",
-                },
-            )
+    def total_requests(self):
+        return len(self.mock.calls)
+
+    def __get_auth_url(self):
+        from bb_wrapper.wrapper.bb import BaseBBWrapper
 
-        self.mocked_auth_requests.post.side_effect = request_auth
+        return BaseBBWrapper()._BaseBBWrapper__oauth_url()
+
+    def set_auth(self, number_of_retries_to_success: int = 0):
+        self.mock.reset()
```

Comment:
não deveria ser um reset apenas no auth?

```suggestion
        self.__get_auth_request().reset()
```

Reply:
Nesse caso é o mock mesmo, porque eu adiciono o callback da autenticação no final da função, aí precisa sobrescrever.